include/zephyr/device.h

```diff
@@ -443,8 +443,16 @@ struct pm_device;
  * @brief Runtime device structure (in ROM) per driver instance
  */
 struct device {
-	/** Name of the device instance */
+#if defined(CONFIG_DEVICE_STORE_NAME) || defined(__DOXYGEN__)
```

Comment:
why `CONFIG_DEVICE_STORE_NAME` can't be set for documentation build instead?

Reply:
I think this pattern is better than keeping an endless/hard to maintain list in Doxyfile, it's now used in a few other places.

Reply:
OK. That's good excuse.